low-level/framework/messaging/ingress_processor.py

```diff
@@ -185,31 +185,44 @@ def _process_msg(self, body):
                     "IngressProcessor, Authentication failed on message: %s" % ingressMsg)
                 return
 
+            # Check for debugging being activated in the message header
+            self._check_debug(message)
+            self._log_debug("_process_msg, ingressMsg: %s" % ingressMsg)
+
             # Get the incoming message type
             if message.get("actuator_request_type") is not None:
                 msgType = message.get("actuator_request_type")
 
                 # Validate against the actuator schema
                 validate(ingressMsg, self._actuator_schema)
+                # Compare hostname from the request to determine
+                # if request is meant for the current node
+                target_node_id = message.get("target_node_id")
+                if target_node_id is None:
+                    logger.warning(
+                        "Required attribute target_node_id is missing "
+                        "from actuator request")
+                    return
+                elif target_node_id == self._node_id:
+                    self._send_to_msg_handler(msgType, message, uuid)
+                else:
+                    logger.info(
+                        "Node identifier mismatch, actuator request rejected.")
+                    return
 
             elif message.get("sensor_request_type") is not None:
                 msgType = message.get("sensor_request_type")
 
                 # Validate against the sensor schema
                 validate(ingressMsg, self._sensor_schema)
+                self._send_to_msg_handler(msgType, message, uuid)
 
             else:
                 # We only handle incoming actuator and sensor requests, ignore
                 # everything else.
                 return
 
-            # Check for debugging being activated in the message header
-            self._check_debug(message)
-            self._log_debug("_process_msg, ingressMsg: %s" % ingressMsg)
-
-            self._send_to_msg_handler(msgType, message, uuid)
-
-        except Exception as ex:
+        except Exception:
```

Comment:
we should always capture exception in a var and printed / passed

Reply:
Actually, it was picked up by Codacy as unused variable. I will add it to the log statement.

Reply:
Added exception variable in log statement.